packages/core/src/Stack/Stack.tsx

```diff
@@ -0,0 +1,74 @@
+import * as React from 'react';
+import { cx } from 'emotion';
+import { PolymorphicProps, PolymorphicComponent } from '../Primitive/Primitive';
+import { useBox } from '../Box';
+import type { MarginProps, PaddingProps, CommonProps, Spacing } from '../types';
+import { Flex, FlexInternalProps } from '../Flex/Flex';
+import { getStackStyles } from './Stack.styles';
+
+export interface StackInternalProps
+  extends CommonProps,
+    MarginProps,
+    PaddingProps,
+    Pick<
+      FlexInternalProps,
+      'isInline' | 'alignItems' | 'justifyContent' | 'flexWrap'
+    > {
+  /**
+   * Defines how flexbox items are ordered within a flexbox container. */
+  flexDirection?: 'row' | 'column';
+  /**
+   * The space between each stack item
+   */
+  spacing?: Spacing;
```

Comment:
is it a `spacing` or a `gap`?
Idk either, I'm just thinking that this works similarly with rowGap, columnGap, and gap in `display: grid`

Naming is the hardest part of coding

Reply:
We have `spacing` in button group already, so I decided to go this way.

Reply:
Oh I think there it's "spacing" hahahahaha

I mean, in my head, gap is like blank space between elements (like the stack here and grid)
and spacing is more what padding does in general

But it was just a comment, I approved the PR
naming is a matter of convention 👍